graphql/resolver.go

```diff
@@ -170,6 +172,43 @@ func New(apiURL string) Config {
 		}
 		return nil, Forbidden.Send(ctx, fmt.Sprintf("user %s does not have permission to access settings for the project %s", user.Username(), projectId))
 	}
+	c.Directives.RequireProjectAccessNew = func(ctx context.Context, obj interface{}, next graphql.Resolver, permission ProjectPermission, access AccessLevel) (interface{}, error) {
+		user := mustHaveUser(ctx)
+
+		args, isMap := obj.(map[string]interface{})
+		if !isMap {
+			return nil, InternalServerError.Send(ctx, "converting args into map")
+		}
+
+		perms, permsOk := projectPermissionAccessMap[permission][access]
```

Comment:
It feels like this map really isn't used for all that much, and seems super likely to get out of date with what the routes use. [Could this function](https://github.com/evergreen-ci/evergreen/blob/6f9165e4b4763a89d24fb4a3750fb42ff49f3c73/globals.go#L1304) be used to return permission levels instead? It seems like then graphql/util could just have a helper function that converts your ProjectPermission values to the evergreen.global ones, as opposed to redefining the mapping.